ethcore/sync/src/chain/mod.rs

```diff
@@ -761,14 +761,24 @@ impl ChainSync {
 						}
 					}
 
-					// Only ask for old blocks if the peer has a higher difficulty
-					if force || higher_difficulty {
+					// Only ask for old blocks if the peer has a higher difficulty than the last imported old block
+					let last_imported_old_block_difficulty = self.old_blocks.as_mut().and_then(|d| {
+						io.chain().block_total_difficulty(BlockId::Number(d.last_imported_block_number()))
+					});
+
+					if force || last_imported_old_block_difficulty.map_or(true, |ld| peer_difficulty.map_or(true, |pd| pd > ld)) {
```

Comment:
In the case of the peer's difficulty being unknown, I was copying the existing behaviour: https://github.com/paritytech/parity-ethereum/blob/82a7ed2f912f22df78ee2417964f82d5a2ecfc43/ethcore/sync/src/chain/mod.rs#L731. So I guess we're being optimistic and sending the request anyway, we're assuming that most of the time we know the peer difficulty.

Not so sure about the case where the imported block diff is not known. That probably would mean (assuming `old_blocks` is Some) that `last_imported_block_number` has not been imported for whatever reason. So what to do in that case? Better to send the request anyway or not? Might be okay since it should almost always be `Some`, and in case it is `None` it might be okay to ask for it again.